quodlibet/quodlibet/ext/events/synchronize_to_device.py

```diff
@@ -0,0 +1,144 @@
+import os
+import string
+from threading import Thread
+
+from gi.repository import Gtk, GLib
+from pydub import AudioSegment
+from quodlibet import _, app
+from quodlibet import config
+from quodlibet import get_user_dir
+from quodlibet import qltk
+from quodlibet.compat import itervalues
+from quodlibet.plugins import PluginConfigMixin
+from quodlibet.plugins.events import EventPlugin
+from quodlibet.qltk.ccb import ConfigCheckButton
+from quodlibet.query import Query
+
+
+class SyncToDevice(EventPlugin, PluginConfigMixin):
+    PLUGIN_ID = "synchronize_to_device"
+    PLUGIN_NAME = _("Synchronize to device")
+    PLUGIN_DESC = _(
+        "Converts all songs from the selected saved searches to 192 kbs and synchronizes them with the specified" 
+        "folder. All songs in that folder, which are not in the saved searches, will be deleted."
+        "Needs the python package 'pydub'.")
+    c_path = __name__ + '_path'
+
+    def PluginPreferences(self, parent):
+        vbox = Gtk.VBox(spacing=6)
+
+        queries = {}
+        query_path = os.path.join(get_user_dir(), 'lists', 'queries.saved')
+        try:
+            with open(query_path, 'rU') as query_file:
+                log = Gtk.TextView()
+                log.set_left_margin(5)
+                log.set_right_margin(5)
+                log.editable = False
+                buffer = log.get_buffer()
+                scroll = Gtk.ScrolledWindow()
+                scroll.set_size_request(-1, 100)
+                scroll.add(log)
+
+                def append(text):
```

Comment:
This is racy, the buffer and log could be destroyed one idle_add calls the callback.

Reply:
How could the buffer and log be destroyed?

Reply:
Do you need to use idle_add here?